tools/image_processing/imagej2/imagej2_bunwarpj_compose_elastic.xml

```diff
@@ -1,25 +1,31 @@
-<?xml version='1.0' encoding='UTF-8'?>
 <tool id="imagej2_bunwarpj_compose_elastic" name="Compose two elastic transformations" version="@WRAPPER_VERSION@.0">
     <description>into a raw transformation with bUnwarpJ</description>
     <macros>
         <import>imagej2_macros.xml</import>
     </macros>
     <expand macro="fiji_requirements" />
-    <command>
-<![CDATA[
-    python $__tool_directory__/imagej2_bunwarpj_compose_elastic.py
-    --target_image "$target_image"
-    --target_image_format $target_image.ext
-    --source_image "$source_image"
-    --source_image_format $source_image.ext
-    --target_elastic_transformation "$target_elastic_transformation"
-    --source_elastic_transformation "$source_elastic_transformation"
-    --output "$output"
-]]>
-    </command>
+    <command detect_errors="exit_code"><![CDATA[
+#import os
+#set source_sans_ext = $os.path.splitext($os.path.basename($source_image.file_name))[0]
+#set source_with_ext = ".".join([source_sans_ext, $source_image.ext])
+ln -s '$source_image.file_name' '$source_with_ext' &&
+#set target_sans_ext = $os.path.splitext($os.path.basename($target_image.file_name))[0]
+#set target_with_ext = ".".join([target_sans_ext, $target_image.ext])
+ln -s '$target_image.file_name' '$target_with_ext' &&
+bunwarpj -compose_elastic
+'$target_with_ext'
+'$source_with_ext'
+'$target_elastic_transformation'
+'$source_elastic_transformation'
+'$output'
+&>output_log.txt;
+if [[ $? -ne 0 ]]; then
+    cp output_log.txt '$output';
```

Comment:
can this if not be removed?

We could add `from_work_dir='output_log.txt'` to the output, isn't it?

Reply:
This "if block" is checking if the bash command ended with an error, and if it does, then it copies the stdout and stderr log to the output dataset so it can be viewed from the browser.

Reply:
If you remove this altogether the output will be automatically on stdout in Galaxy under the `i` tab in Galaxy. Or do you only want to save this when the tool crashed?

Reply:
The problem with Have is that it produces a lot of logging, and the first portion of this is displayed in the history item unless stdout is redirected.  If there is a better way to do this, let me know and I'll change the approach.